airflow/api_fastapi/execution_api/datamodels/asset.py

```diff
@@ -34,3 +34,18 @@ class AssetAliasResponse(BaseModel):
 
     name: str
     group: str
+
+
+class AssetProfile(BaseModel):
+    """
+    Profile of an Asset.
+
+    Asset will have name, uri and asset_type defined.
+    AssetNameRef will have name and asset_type defined.
+    AssetUriRef will have uri and asset_type defined.
+
+    """
+
+    name: str | None = None
+    uri: str | None = None
+    asset_type: str
```

Comment:
I realised there was a bug in my earlier code where i was defining "asset_type" at the payload level. The issue with that is if you have more than 1 outlet passed, it will always take the "asset_type" of the last outlet and not register events for the past ones.

Reply:
Fixed.